### PR TITLE
core/tracker: add sync duties to tracker

### PR DESCRIPTION
**Pull request by dB2510**
Adds sync committee duties support to tracker.

category: feature
ticket: #1298

### CODE REVIEW

File: core/tracker/tracker_internal_test.go [hunk 1]
```diff
@@ -648,125 +648,198 @@ func TestFromSlot(t *testing.T) {
 	<-done
 }
 
-func TestAnalyseDutyFailedAgg(t *testing.T) {
+func TestAnalyseFetcherFailed(t *testing.T) {
```
Comment (dB2510) [inline core/tracker/tracker_internal_test.go:651]: converted these tests to table tests

File: core/tracker/tracker.go [hunk 2]
```diff
@@ -320,6 +342,38 @@ func analyseFetcherFailed(duty core.Duty, allEvents map[core.Duty][]event) (bool
 		return false, fetcher, ""
 	}
 
+	// Duty sync contribution depends on prepare sync contribution duty, so check if that was why it failed.
+	if duty.Type == core.DutySyncContribution {
```
Comment (dB2510) [inline core/tracker/tracker.go:346]: added cases for DutySyncContribution too

File: core/tracker/tracker.go [hunk 3]
```diff
@@ -400,10 +463,20 @@ func isParSigEventExpected(duty core.Duty, pubkey core.PubKey, allEvents map[cor
...
+		return scheduled(core.DutySyncContribution)
+	}
+
 	// For all other duties check if the type itself was scheduled.
 	return scheduled(duty.Type)
 }
 
+// canSchedule returns true if the given duty type can be scheduled by scheduler.
+func canSchedule(duty core.DutyType) bool {
+	return !(duty == core.DutyExit || duty == core.DutyBuilderRegistration || duty == core.DutySyncMessage)
```
Comment (corverroos) [inline core/tracker/tracker.go:477]: `duty != core.DutyExit && duty != core.DutyBuilderRegistration && duty != core.DutySyncMessage`

File: core/tracker/tracker.go [hunk 4]
```diff
@@ -400,10 +463,20 @@ func isParSigEventExpected(duty core.Duty, pubkey core.PubKey, allEvents map[cor
 		return scheduled(core.DutyAttester)
 	}
 
+	// For DutyPrepareSyncContribution, check that if DutySyncContribution was scheduled.
+	if duty.Type == core.DutyPrepareSyncContribution {
+		return scheduled(core.DutySyncContribution)
```
Comment (corverroos) [inline core/tracker/tracker.go:468]: `DutySyncMessage`

**Review by corverroos — APPROVED** [review]

File: core/tracker/tracker.go (hunk 4 above)
Comment (dB2510) [reply core/tracker/tracker.go:468]: `DutySyncMessage` isn't scheduled by scheduler

**Pull request merged**